src/System.CommandLine.Hosting/HostingExtensions.cs

```diff
@@ -64,5 +66,31 @@ public static class HostingExtensions
                     services.Configure(configureOptions);
             });
         }
+
+        public static InvocationContext GetInvocationContext(this IHostBuilder hostBuilder) =>
```

Comment:
If `hostBuilder` is `null`, this should throw an `ArgumentNullException`. Same comment for `GetInvocationContext`.

Reply:
For example, from aspnet/Extensions: https://github.com/aspnet/Extensions/blob/07021213ce1947a910570263eb543f6fe01e7e09/src/Options/Options/src/ConfigureOptions.cs#L32

Reply:
https://github.com/dotnet/command-line-api/pull/664/commits/2449ac12c33a3f2a10450dd8e6be4e89d2af19eb